### DISCUSSION TITLE
CORS error on NextJS API route

### DISCUSSION THREAD

**Discussion by sandervspl** [vercel/community#1401] [Help] [answered]
My production website (NextJS 13) can not fetch from its own API routes because of CORS errors. I have added the wildcard for allowed origins, and all the tests indicate this is working except for the website itself. On all tests I see the allowed origin header return the wildcard, but in the browser on my website I don't see the header in the response.

I have recently (yesterday) transfered domains from Cloudflare to Vercel, so I think something is going wrong because of that, but I don't know for sure. I have wiped everything from Cloudflare, so that should be sorted.

The response status is also 308 which I find strange.

You can see the request failing here  

https://www.wowvalor.app/en/death-knight/blood/mm#top-talents

According to this website everything should be working correctly CORS-wise https://cors-test.codehappy.dev/?url=https%3A%2F%2Fwowvalor.app%2Fapi%2Fblizzard%2Ftalent-tree%2Fshaman%2Felemental&origin=https%3A%2F%2Fwww.wowvalor.app&method=get

**Comment by sandervspl** [comment, marked as answer]
I fixed it.

I was using `wowvalor.app` instead of `www.wowvalor.app` in the request URL. Seems like the redirect from wowvalor.app -> www.wowvalor.app is somehow broken for fetch requests.

**Comment by untilhamza** [comment]
For anyone struggling with this, refer to this answer on stack overflow
https://stackoverflow.com/a/76505443/13889098
